File: backend/member-service/src/main/java/com/example/memberservice/global/decoder/FeignErrorDecoder.java
package com.example.memberservice.global.decoder;

import com.example.memberservice.global.exception.CustomException;
import com.example.memberservice.global.exception.ErrorCode;
import feign.Response;
import feign.codec.ErrorDecoder;

public class FeignErrorDecoder implements ErrorDecoder {

    @Override
    public Exception decode(String methodKey, Response response) {
        switch (response.status()){
            case 400:
                return new CustomException(ErrorCode.USERID_NOT_FOUND, "");
            case 404:
                if(methodKey.contains("UserId")){
                    return new CustomException(ErrorCode.USERID_NOT_FOUND_ERROR, "");
                }
                break;
            default:
                return new Exception(response.reason());
        }
        return null;
    }
}

File: backend/member-service/src/main/kotlin/com/example/memberservice/global/decoder/FeignErrorDecoder.kt
package com.example.memberservice.global.decoder

import com.example.memberservice.global.exception.CustomException
import com.example.memberservice.global.exception.ErrorCode
import feign.Response
import feign.codec.ErrorDecoder

class FeignErrorDecoder : ErrorDecoder {

    override fun decode(methodKey: String, response: Response): Exception? {
        when (response.status()) {
            400 -> return CustomException(ErrorCode.USERID_NOT_FOUND, "")
            404 -> if (methodKey.contains("UserId")) {
                return CustomException(ErrorCode.USERID_NOT_FOUND_ERROR, "")
            }
            else -> return Exception(response.reason())
        }
        return null
    }
}
